content/blog/D0001-openage_mod_api_game_entity.md

```diff
@@ -0,0 +1,57 @@
+
+Developers and modders should not let their objects inherit directly from `GameEntity` and instead choose one of the four child objects `Unit`, `Ambient`, `Item` or `Building`. With the exception of `Building` these objects are just for categorization and don't require additional attributes to be defined. An `Item` object could have the same abilities as a `Unit` object \*. However the categorization helps us, the engine developers, by being able to design abilities that are aimed at a specific category. For example, the `PickupItem` will first and foremost be designed to handle game entities of type `Item`. We'll now explain what is expected from each category.
+
+\* Fun fact: Because nyan allows for multiple inheritance an object can be in more than one of the categories. So if you think that your units should also be items, you cango for it!
+
+## Unit
+
+`Unit` is probably the most generic category. In general, all game entities that move, die, decay, fly and have HP should be units. In the current API design, units are the only types of `GameEntity` that can be trained with the `Train` ability. 
+
+Examples: military, villagers, animals
+
+## Ambient
+
+`Ambient` objects usually lie or stand around on the map and look pretty. Typically, these objects never move, do not have HP and some of them are harvestable for resources. Flags and flag attachements would also be classified as `Ambient`. In AoE2, most of these objects would be owned by the neutral player *Gaia*.
+
+Examples: cactus, tree, gold/stone pile, mountain
+
+## Item
+
+These are very similar to `Ambient` objects, but can be put in an inventory. openage features an inventory system through the `Inventory` ability. Items in there can have special effects on the inventory owner, provide boni or grant new abilities.
+
+Example: relic
+
+## Building
+
+![Building](../images/D0001-building.png)
+
+Buildings are more sophisticated than the othere three categories because they have to be constructed first and need space on the map grid. Each `Building` can have construction and damage stages that are defined in a `Progress` object. `Progress` objects allow buildings to display an alternative sprite after a certain percentage of *something* is complete. The number of stages is unlimited and theoretically a building could have 100 stages of construction that each have their own graphic.
+
+It is also allowed to define more complex buildings that consist of multiple parts by using `MultiPartBuilding`. This type acts as a container that manages several subbuildings that are allowed to have individual properties, e.g. certain abilities and boni. `MultiPartBuildings` are also useful for modelling buildings that have passable parts like the Town Center in AoE2.
```

Comment:
how is passability for building parts defined? (so that one can walk below the town center roofs but not through the center building)

Reply:
It's an ability (`Passable`) with no members now. I moved a few members of `Building` into abilities yesterday. Will fix it in the blogpost asap